qiskit/primitives/statevector_sampler.py

```diff
@@ -48,86 +48,77 @@ class _MeasureInfo:
     qreg_indices: list[int]
 
 
-class Sampler(BaseSamplerV2):
+class StatevectorSampler(BaseSamplerV2):
     """
     Simple implementation of :class:`BaseSamplerV2` with Statevector.
     """
 
-    _DEFAULT_SHOTS: int = 512
-
-    def __init__(self, *, seed: np.random.Generator | int | None = None):
+    def __init__(self, *, default_shots: int = 1024, seed: np.random.Generator | int | None = None):
         """
         Args:
-            seed: The seed for random number generator.
+            default_shots: The default shots for the sampler if not specified during run.
+            seed: The seed or Generator object for random number generation.
+                If None, a random seeded default RNG will be used.
         """
+        self._default_shots = default_shots
         self._seed = seed
-        if isinstance(self._seed, np.random.Generator):
-            self._rng = self._seed
-        else:
-            self._rng = np.random.default_rng(self._seed)
 
     @property
-    def seed(self) -> np.random.Generator | int | None:
-        """Return the seed for random number generator.
+    def default_shots(self) -> int:
```

Comment:
I'm wondering whether we need the setter of `default_shots`?

Reply:
I think we should be fine to leave it off; they can make a new sampler, or use the run args.